dataUtil/featureExtraction.py

```diff
@@ -0,0 +1,114 @@
+from librosa.feature import mfcc
+from librosa.display import specshow
+import matplotlib.pyplot as plt
+
+
+def get_fft(signal, sampling_frequency):
+    '''
+    This function takes in an array, either of size (num_frames,) or (num_batches, num_frames)
+        and returns a tuple of times, frequencies, and a spectrogram.
+    :param signal: the signal to get the spectrogram from
+    :param sampling_frequency: the sampling frequency of the signal
+    :param window_size: the number of samples per time segment (column) of the spectrogram
+    :return: a tuple of segment times (shape: [num_batches,] num_segments),
+                        sample frequncies (shape: [num_baches,] num_frequencies),
+                        spectrogram (shape: [num_batches,] num_segments, num_frequencies)
+    '''
+    frequencies, times, spectrogram = spectro(signal, fs=sampling_frequency, nfft=2048)
+    # Only return important frequencies
+    return times, frequencies[:70], np.swapaxes(spectrogram, 1, 2)[:, :, :70]
+
+
+def _get_single_mfcc(signal, sampling_frequency, num_mfcc):
+    '''
+    This function takes in an array of size (num_frames,) and returns the mfccs of size (num_segments, num_mfcc)
+    :param signal: the signal to get the mfccs from
+    :param sampling_frequency: the sampling frequency of the signal
+    :param num_mfcc: the number of mfccs to extract
+    :return: an array of size (num_segments, num_mfcc)
+    '''
+    return mfcc(y=signal, sr=sampling_frequency, n_mfcc=num_mfcc).T
+
+def get_mfcc(signals, sampling_frequency, num_fcc):
+    '''
+    This function takes in a batch of signals of size (num_batches, num_frames,)
+        and returns the mfccs of size (num_batches, num_segments, num_mfcc)
+    :param signals: the signal to get the mfccs from
+    :param sampling_frequency: the sampling frequency of the signal
+    :param num_fcc: the number of mfccs to extract
+    :return: an array of size (num_batches, num_segments, num_mfcc)
+    '''
+    mfccs = []
+    for signal in signals:
+        mfccs.append(_get_single_mfcc(signal, sampling_frequency, num_fcc))
+    return np.array(mfccs)
+
+def plot_spectrogram(times, frequencies, spectrogram):
+    '''
+    Plot the spectrogram.
+    :param times: an array of size (num_segments,)
+    :param frequencies: an array of size (num_frequencies,)
+    :param spectrogram: a spectrogram of size (num_segments, num_frequencies)
+    :return: None
+    '''
+    plt.pcolormesh(times, frequencies, spectrogram.T)
+    plt.ylabel("Frequency [Hz]")
+    plt.xlabel("Time [Sec]")
+    plt.show()
+
+
+def plot_mfcc(mfcc):
+    '''
+    Plot the mfccs.
+    :param mfcc: an array of size (num_segments, num_mfcc)
+    :return: None
+    '''
+    specshow(mfcc.T, x_axis='time')
+    plt.colorbar()
+    plt.title("MFCC")
+    plt.tight_layout()
+    plt.show()
+
+
+def main():
+    a = read_audio_data("../data/processed/english/english.npy").astype(np.float32)
```

Comment:
Why are we going up one directory? Shouldn’t the current working directory be set such that all references are relative to the root directory (Accenter)?